### DISCUSSION TITLE
Why do in-memory realms spawn a file-extension-less realm file on windows desktop?

### DISCUSSION THREAD

**Discussion by daawaan4x** [realm/realm-dotnet#2722] [Q&A] [answered]
Title pretty much sums up everything I want to say 😄 getting a realm instance with an `InMemoryConfiguration` creates a realm file in the documents folder here on Windows 10.

**Comment by nirinchev** [comment, marked as answer]
This file is used as swap in case the data in the Realm can't fit in the computer's memory. Most of the time it should be completely empty though, unless you're working with massive amounts of data. It should also be deleted when you close the last instance referencing it.

[Here's a similar issue](https://github.com/realm/realm-cocoa/issues/2029) in the Cocoa repository where Alex and JP go into a little bit more detail.
